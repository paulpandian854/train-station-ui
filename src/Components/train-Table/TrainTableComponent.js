function TrainTableComponent(props) {
    return (
        <div data-testid="train-1">{props.jwt}</div>
    )

}

 export const functionForTesting = (a,b) => {
     return ( a + b);
 }

export default TrainTableComponent;