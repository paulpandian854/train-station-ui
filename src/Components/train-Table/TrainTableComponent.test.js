
import TrainTableComponent from './TrainTableComponent';
import {render, screen, cleanup} from '@testing-library/react';

import renderer from 'react-test-renderer';
afterEach(cleanup);
//Mock Function
const functionForTesting = jest.fn((x,y)=> 3)

test('Test TrainTable Component', () => {
    expect(1).toBe(1);
})

test('Test TrainTable Component and props', () => {
    render(<TrainTableComponent jwt="abcd"/>);
    const traindetails = screen.getByTestId('train-1');
    expect(traindetails).toBeInTheDocument();
    expect(traindetails).toHaveTextContent('abcd');
})

test('Test TrainTable Component3', () => {
    render(<TrainTableComponent jwt="abcd"/>);
    const traindetails = screen.getByTestId('train-1');
    expect(traindetails).toBeInTheDocument();
    expect(traindetails).toHaveTextContent('abcd');
})


test('Test TrainTable Componen- Snapshot testing', () => {
    const tree = renderer.create(<TrainTableComponent jwt="abcd"/>).toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
})

test('Test a dummy function inside train details component', () => {
    expect(functionForTesting(1,2)).toBeUndefined();
})