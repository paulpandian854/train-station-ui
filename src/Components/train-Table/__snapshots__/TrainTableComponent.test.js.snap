// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TrainTable Componen- Snapshot testing 1`] = `
<div
  data-testid="train-1"
>
  abcd
</div>
`;
